yb-voyager/cmd/common.go

```diff
@@ -1191,6 +1192,40 @@ func ParseJSONToAssessmentReport(reportPath string) (*AssessmentReport, error) {
 	return &report, nil
 }
 
+// Implementing json.Marshaler interface by defining MarshalJSON() to have customised serialization
+// NOTE: Ensure that AssessmentReport is passed as value in json.Marshal() to invoke custom MarshalJSON logic.
+func (ar AssessmentReport) MarshalJSON() ([]byte, error) {
+	switch ar.SerializationMode {
+	case "full", "":
+		type Alias AssessmentReport
+		return json.Marshal(Alias(ar))
```

Comment:
For YBD payload, do we need to send the new fields we are adding, given that we will have to send them separately in YBD assess-migration payload with payload version change. If not required, we can remove them from her to reduce the payload size as it will double in size and will be significant in case of `Issues` as these will be separately in hight payload and also in this `AssessmentReport` struct

Reply:
Decided to live with this for now. Will be cleaned up in future.